src/charm.py

```diff
@@ -108,6 +113,31 @@ def _configure(self):
 
         self.unit.status = ActiveStatus()
 
+    def _set_alerts(self, container):
+        """Create alert rule files for all Prometheus consumers.
+
+        Args:
+            container: the Prometheus workload container into which
+                alert rule files need to be created.
+        """
+        if not self.provider_ready:
+            return
+
+        with container.is_ready():
+            container.remove_path(RULES_DIR, recursive=True)
+
+        for rel_id, alerts in self.prometheus_provider.alerts().items():
```

Comment:
shouldn't the if block be under the context manager as well, instead of the remaining try-except?

Reply:
which "if block" ?

Reply:
I am trapping `ConnectionError` on a per alert rule basis.